### ISSUE TITLE
BBC: Blead Breaks indirect

### ISSUE THREAD

**Issue by cjg-cguevara** [Perl/perl5#17362]
This is a bug report for perl from "Carlos Guevara" <carlos@carlosguevara.com>,
generated with the help of perlbug 1.41 running under perl 5.31.7.

-----------------------------------------------------------------
[Please describe your issue here]

BBC: Blead Breaks indirect

Please see http://matrix.cpantesters.org/?dist=indirect

[Please do not change anything below this line]
-----------------------------------------------------------------
---
Flags:
    category=core
    severity=low
---
Site configuration information for perl 5.31.7:

Configured by cpan at Wed Dec 11 21:38:33 CST 2019.

Summary of my perl5 (revision 5 version 31 subversion 7) configuration:
  Commit id: f34962468be5ced287b17b0844c31ee180ffe736
  Platform:
    osname=linux
    osvers=4.9.0-11-amd64
    archname=x86_64-linux-thread-multi
    uname='linux cjg-stretch 4.9.0-11-amd64 #1 smp debian 4.9.189-3+deb9u2 (2019-11-11) x86_64 gnulinux '
    config_args='-des -Dprefix=~/bin/perl-blead -Dscriptdir=~/bin/perl-blead/bin -Dusedevel -Duse64bitall -Duseithreads'
    hint=recommended
    useposix=true
    d_sigaction=define
    useithreads=define
    usemultiplicity=define
    use64bitint=define
    use64bitall=define
    uselongdouble=undef
    usemymalloc=n
    default_inc_excludes_dot=define
    bincompat5005=undef
  Compiler:
    cc='cc'
    ccflags ='-D_REENTRANT -D_GNU_SOURCE -fwrapv -fno-strict-aliasing -pipe -fstack-protector-strong -I/usr/local/include -D_LARGEFILE_SOURCE -D_FILE_OFFSET_BITS=64 -D_FORTIFY_SOURCE=2'
    optimize='-O2'
    cppflags='-D_REENTRANT -D_GNU_SOURCE -fwrapv -fno-strict-aliasing -pipe -fstack-protector-strong -I/usr/local/include'
    ccversion=''
    gccversion='6.3.0 20170516'
    gccosandvers=''
    intsize=4
    longsize=8
    ptrsize=8
    doublesize=8
    byteorder=12345678
    doublekind=3
    d_longlong=define
    longlongsize=8
    d_longdbl=define
    longdblsize=16
    longdblkind=3
    ivtype='long'
    ivsize=8
    nvtype='double'
    nvsize=8
    Off_t='off_t'
    lseeksize=8
    alignbytes=8
    prototype=define
  Linker and Libraries:
    ld='cc'
    ldflags =' -fstack-protector-strong -L/usr/local/lib'
    libpth=/usr/local/lib /usr/lib/gcc/x86_64-linux-gnu/6/include-fixed /usr/include/x86_64-linux-gnu /usr/lib /lib/x86_64-linux-gnu /lib/../lib /usr/lib/x86_64-linux-gnu /usr/lib/../lib /lib
    libs=-lpthread -lnsl -ldl -lm -lcrypt -lutil -lc
    perllibs=-lpthread -lnsl -ldl -lm -lcrypt -lutil -lc
    libc=libc-2.24.so
    so=so
    useshrplib=false
    libperl=libperl.a
    gnulibc_version='2.24'
  Dynamic Linking:
    dlsrc=dl_dlopen.xs
    dlext=so
    d_dlsymun=undef
    ccdlflags='-Wl,-E'
    cccdlflags='-fPIC'
    lddlflags='-shared -O2 -L/usr/local/lib -fstack-protector-strong'

---
@INC for perl 5.31.7:
    /home/cpan/bin/perl-blead/lib/site_perl/5.31.7/x86_64-linux-thread-multi
    /home/cpan/bin/perl-blead/lib/site_perl/5.31.7
    /home/cpan/bin/perl-blead/lib/5.31.7/x86_64-linux-thread-multi
    /home/cpan/bin/perl-blead/lib/5.31.7

---
Environment for perl 5.31.7:
    HOME=/home/cpan
    LANG=en_US.UTF-8
    LANGUAGE (unset)
    LC_ALL=C
    LD_LIBRARY_PATH (unset)
    LOGDIR (unset)
    PATH=/home/cpan/bin/perl-blead/bin:/home/cpan/bin:/home/cpan/bin:/usr/local/bin:/usr/bin:/bin:/usr/local/games:/usr/games
    PERL_BADLANG (unset)
    SHELL=/bin/bash

**Comment by jkeenan** [comment]
> This is a bug report for perl from "Carlos Guevara" [carlos@carlosguevara.com](mailto:carlos@carlosguevara.com),
> generated with the help of perlbug 1.41 running under perl 5.31.7.
> 
> [Please describe your issue here]
> 
> BBC: Blead Breaks indirect
> 
> Please see http://matrix.cpantesters.org/?dist=indirect
[snip]
> Summary of my perl5 (revision 5 version 31 subversion 7) configuration:
> Commit id: [f349624](https://github.com/Perl/perl5/commit/f34962468be5ced287b17b0844c31ee180ffe736)
> Platform:
> osname=linux
> osvers=4.9.0-11-amd64
> archname=x86_64-linux-thread-multi
> config_args='-des -Dprefix=~/bin/perl-blead -Dscriptdir=~/bin/perl-blead/bin -Dusedevel -Duse64bitall -Duseithreads'
[snip]

This breakage is, AFAICT, only occuring on threaded builds.  Bisecting with this invocation:
```
perl Porting/bisect.pl -Duseithreads \
--start=1c5f8ad8fa5c2c65862926c21a3989a4f5c59845 \
--end=f34962468be5ced287b17b0844c31ee180ffe736 \
--module=indirect
```
... points to this commit:
```
commit e6c7056ba6f1dca6d04e6d36515a0ffc3a5ec02a
Author:     Stefan Seifert <nine@detonation.org>
AuthorDate: Wed Oct 30 16:23:01 2019 +0100
Commit:     Tony Cook <tony@develop-help.com>
CommitDate: Thu Dec 12 11:35:20 2019 +1100

    Move PL_check to the interp vars to fix threading issues
    
    Fixes issue #14816

```

(Bisecting without `-Duseithreads` got no failures.)

@tonycoz, @niner, can you take a look?

Thank you very much.
Jim Keenan

**Comment by Leont** [comment]
This is this issue @nwc10 describes in #14816

**Comment by jkeenan** [comment]
> > This is a bug report for perl from "Carlos Guevara" [carlos@carlosguevara.com](mailto:carlos@carlosguevara.com),
> > generated with the help of perlbug 1.41 running under perl 5.31.7.
> > [Please describe your issue here]
> > BBC: Blead Breaks indirect
> > Please see http://matrix.cpantesters.org/?dist=indirect
> > [snip]
> > Summary of my perl5 (revision 5 version 31 subversion 7) configuration:
> > Commit id: [f349624](https://github.com/Perl/perl5/commit/f34962468be5ced287b17b0844c31ee180ffe736)
> > Platform:
> > osname=linux
> > osvers=4.9.0-11-amd64
> > archname=x86_64-linux-thread-multi
> > config_args='-des -Dprefix=~/bin/perl-blead -Dscriptdir=~/bin/perl-blead/bin -Dusedevel -Duse64bitall -Duseithreads'
> > [snip]
> 
> This breakage is, AFAICT, only occuring on threaded builds. Bisecting with this invocation:
> 
> ```
> perl Porting/bisect.pl -Duseithreads \
> --start=1c5f8ad8fa5c2c65862926c21a3989a4f5c59845 \
> --end=f34962468be5ced287b17b0844c31ee180ffe736 \
> --module=indirect
> ```
> 
> ... points to this commit:
> 
> ```
> commit e6c7056ba6f1dca6d04e6d36515a0ffc3a5ec02a
> Author:     Stefan Seifert <nine@detonation.org>
> AuthorDate: Wed Oct 30 16:23:01 2019 +0100
> Commit:     Tony Cook <tony@develop-help.com>
> CommitDate: Thu Dec 12 11:35:20 2019 +1100
> 
>     Move PL_check to the interp vars to fix threading issues
>     
>     Fixes issue #14816
> ```
> 
> (Bisecting without `-Duseithreads` got no failures.)
> 
> @tonycoz, @niner, can you take a look?
> 
> Thank you very much.
> Jim Keenan

In this commit:
```
commit 0850138d5ff43e9778637c50e5dfc4c06ada52c6 
Author:     Tony Cook <tony@develop-help.com>
AuthorDate: Mon Dec 16 08:56:52 2019 +1100
Commit:     Tony Cook <tony@develop-help.com>
CommitDate: Mon Dec 16 08:57:58 2019 +1100

    Revert "Move PL_check to the interp vars to fix threading issues"
    
    and the associated commits, at least until a way to make
    wrap_op_checker() work is available.
```
... Tony has reverted the commit which caused the BBC for indirect on threaded builds.  Confirmed on both Linux and FreeBSD-13.  So, while there are larger issued in #14816 still to be resolved, this ticket is closable.

Thank you very much.
Jim Keenan
